pkg/plugins/manifest.go

```diff
@@ -83,51 +97,127 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 }
 
 // getPluginSignatureState returns the signature state for a plugin.
-func getPluginSignatureState(log log.Logger, plugin *PluginBase) PluginSignature {
+func getPluginSignatureState(log log.Logger, plugin *PluginBase) (*PluginSignatureState, error) {
```

Comment:
Yeah just because now with private plugins from https://github.com/grafana/grafana/pull/29240/files#diff-798baa53157b806c13e80304f88da52846c1dd1b02cc79497ac50a2dc722873fR128, there are different error cases to handle. Fine for you?

Reply:
@wbrowne You mean so you can return `nil` here in case of an error? I don't really understand if there might be anything else. If that's the case, you can also just return `PluginSignatureState{}`, noone should care about the non-error return value if an error is returned.

Reply:
Precisely. I personally prefer the `nil`, also due to the fact the pattern is already followed in this file

Reply:
Fair points. In that case, I'll revert and bear the same logic in mind for the future 😄

Reply:
Thanks :) It's no big deal in this case, but it's important to bear in mind that pointers and value types have fundamentally different characteristics. For example, a pointer introduces the risk of a nil pointer panic, which the corresponding value type is safe from. I think pointers are mostly useful when you think there's a performance decrease from copying values, or you need to change the underlying value when passing arguments around.